sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -135,6 +137,27 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
    */
   struct TransportOptions final
   {
+#if !defined(BUILD_TRANSPORT_CUSTOM_ADAPTER)
+#if defined(BUILD_TRANSPORT_WINHTTP_ADAPTER) || defined(BUILD_CURL_HTTP_TRANSPORT_ADAPTER)
+    /**
+     * @brief Http Proxies used when making an HTTP connection.
+     *
+     * @remark The URL for the proxy server to use for this connection.
+     */
+    std::string HttpProxy;
+
+    /**
+     * @brief The username to use when authenticating with the proxy server.
+     */
+    std::string ProxyUserName;
+
+    /**
+     * @brief The password to use when authenticating with the proxy server.
+     */
+    std::string ProxyPassword;
```

Comment:
Well, I see your are betting on the `common transport options`.

Remember that building a customer adapter doesn't mean you can't build curl/winHttp as well.

You can build your own transport and you can still build libcurl. (passing both flags)

Reply:
I'm not betting on anything. I implemented common transport options AND transport specific options. The transport specific options are idiomatic to the transport, the common options are common.

If we decide to not do the common transport options, I can always throw this code away and push the problem to the caller.

FWIW, the code in the `TransportOptions` constructor treats custom, libcurl and winhttp as exclusive options. That continues to be the case.